Fix/nex 1989/feature flag undefined variable (#2694)
oat-sa/tao-core: README.md

tao-core
========

[Task Queue Doc](models/classes/taskQueue/README.md)

# Feature flag

> You can now define feature flag in `.ENV` global variables. 

Reviewer: ```suggestion
You can now define feature flag as environment variables. 
```

Cause `.ENV` is a bit misleading. People might thing you need a .env file or something.

Reviewer: It is also important to state that the standard is to have the prefix `FEATURE_FLAG_XXXXX` at some part. Wdyt?

## How to use it

You can use feature flag to easly switch off/on parts of TAO functionality.

### AbstractFeatureFlagFormPropertyMapper

To hide/show form fields you can extend `AbstractFeatureFlagFormPropertyMapper` with your configurable service that has
`OPTION_FEATURE_FLAG_FORM_FIELDS` mapped to list of fields that you want to hide/show. 

```
FeatureFlagFormPropertyMapper::OPTION_FEATURE_FLAG_FORM_FIELDS => [
    formField_01 => [
        FEATURE_FLAG_NAME
    ]

> ]

Reviewer: Maybe use the php configuration file content to be more precise? Wdyt?

```

This configuration will display `formField_01` when `FEATURE_FLAG_NAME` is enabled

### SectionVisibilityFilterInterface

`tao/SectionVisibilityFilter` is responsible for listing sections that can be disabled from user. In order to add more 

> sections that has to be disabled/enabled based on feature flag. 

Reviewer: ```suggestion
sections that have to be disabled/enabled based on feature flag. 
```

```
'featureFlagSections' => [
    'sectionName' => [
        `FETURE_FLAG_01`
    ]
]

> ``` 

Reviewer: Maybe use the php configuration file content to be more precise? Wdyt?
